paasta_tools/kubernetes_tools.py

```diff
@@ -2647,6 +2658,7 @@ def force_delete_pods(
         )
 
 
+@time_cache(ttl=60)
 def get_all_namespaces(
```

Comment:
any chance that returning a set here makes a meaningful impact on speed now that we're calling this in ensure_namespace?

Reply:
The line that checks membership doesn't show up at all in my py-spy traces, so probably not.